utils/common/src/funTest/kotlin/ArchiveUtilsFunTest.kt

```diff
@@ -49,25 +49,14 @@ class ArchiveUtilsFunTest : WordSpec() {
         outputDir = tempdir(testCase.name.testName.replace(' ', '-'))
     }
 
-    /**
-     * Copy the test archive with the given [sourceName] under an unknown file extension to a temporary folder.
-     * This is used to test whether archives with unknown file extensions can be unpacked.
-     */
-    private fun copyTestArchive(sourceName: String): File {
-        val sourcePath = getAssetFile(sourceName)
-        val targetPath = tempdir().resolve("unknown.dat")
-
-        return sourcePath.copyTo(targetPath)
-    }
-
     init {
         // The test for DEB is implemented differently and cannot be covered by the parameterized loop.
         (ArchiveType.entries - ArchiveType.DEB - ArchiveType.NONE).forEach { type ->
             val extension = type.extensions.first()
 
             "${type.name} archives" should {
                 "unpack" {
-                    val archive = getAssetFile("test$extension")
+                    val archive = extractResource("/test$extension", tempfile(suffix = extension))
```

Comment:
If this is going to be used in more places in future it could be useful to improve `extractResource` to automatically keep the extension for the `tempfile` by default.